ime/voiceime/src/main/java/com/google/android/voiceime/ActivityHelper.java

```diff
@@ -63,7 +63,14 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         && data != null
         && data.hasExtra(RecognizerIntent.EXTRA_RESULTS)) {
       ArrayList<String> results = data.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS);
-      createResultDialog(results.toArray(new String[results.size()])).show();
+
+      if(results.size() > 1) {
+        createResultDialog(results.toArray(new String[results.size()])).show();
+      } else if(results.size() > 0) {
```

Comment:
For readability, I would change this to `== 1`
```suggestion
      } else if(results.size() == 1) {
```

Reply:
Updated